docs/runtime/go-metrics.md

```diff
@@ -0,0 +1,221 @@
+  - [Metric: `go.memory.allocations`](#metric-gomemoryallocations)
+- [Go Garbage Collection](#go-garbage-collection)
+  - [Metric: `go.memory.gc.goal`](#metric-gomemorygcgoal)
+- [Go Goroutines](#go-goroutines)
+  - [Metric: `go.goroutine.count`](#metric-gogoroutinecount)
+- [Go Threads](#go-threads)
+  - [Metric: `go.thread.limit`](#metric-gothreadlimit)
+- [Go Scheduler](#go-scheduler)
+  - [Metric: `go.schedule.duration`](#metric-goscheduleduration)
+- [Go Runtime Configuration](#go-runtime-configuration)
+  - [Metric: `go.config.gogc`](#metric-goconfiggogc)
+
+<!-- tocstop -->
+
+## Go Memory
+
+**Description:** Go runtime metrics captured under the namespace `go.memory.*`
+
+### Metric: `go.memory.used`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.go.memory.used(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.memory.used` | UpDownCounter | `By` | Memory used by the Go runtime. [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+**[1]:** Computed from `(/memory/classes/total:bytes - /memory/classes/heap/released:bytes)`.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.used(full) -->
+| Attribute  | Type | Description  | Examples  | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) | Stability |
+|---|---|---|---|---|---|
+| [`go.memory.type`](/docs/attributes-registry/go.md) | string | The type of memory. | `other`; `stack` | `Recommended` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+`go.memory.type` has the following list of well-known values. If one of them applies, then the respective value MUST be used; otherwise, a custom value MAY be used.
+
+| Value  | Description | Stability |
+|---|---|---|
+| `stack` | Memory allocated from the heap that is reserved for stack space, whether or not it is currently in-use. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `other` | Memory used by the Go runtime, excluding other categories of memory usage. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+<!-- endsemconv -->
+
+### Metric: `go.memory.released`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.go.memory.released(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.memory.released` | UpDownCounter | `By` | Memory that is completely free and has been returned to the underlying system. [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+**[1]:** Computed from `/memory/classes/heap/released:bytes`.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.released(full) -->
+<!-- endsemconv -->
+
+### Metric: `go.memory.limit`
```

Comment:
As per [this thread](https://github.com/open-telemetry/semantic-conventions/pull/981#discussion_r1586600729), I think it would be better to have this under `go.config.memlimit`. 

The reason for this as follows:

1. It's a soft rather than hard limit. This clashes with common CNCF nomenclature in k8s where a memory limit is the point where a container gets killed or evicted.
2. What if Go introduces a hard limit in the future?
3. Having all config options under config makes the most sense to me.

I don't feel strongly about this, so feel free to ignore.

Reply:
WDYT of `go.memory.soft_limit` to mitigate points 1 and 2?  I also don't feel strongly, so we can leave this open to see if others have thoughts.

Reply:
Raised this in the SIG meeting.  Additional thoughts/feedback from others mostly supported `go.memory.limit`:

* Don't love `soft_limit`, since people might be confused that it is named differently from GOMEMLIMIT. "I didn't know Go had a soft limit?  Is that the same as GOMEMLIMIT?"
* We could make the new metric `hard_limit`, if a hard limit was introduced in the future.
* How likely is it actually that a hard limit is added?
* Given the configuration is GOMEMLIMIT, `go.memory.limit` seems reasonably safe.  Go would need to choose something different e.g. GOHARDMEMLIMIT, etc to introduce a new limit anyways.

Reply:
Your last comment mitigates my worries about 2. 1 and 3 still apply tho. Was the `go.config.memlimit` option I suggested discussed in the SIG meeting?

Anyway, I don't feel very strongly about this, so feel free to go ahead with `go.memory.limit` if you think it has some consensus behind it.

Reply:
I raised it as one of the options.  I don't believe there were any specific objections, but that it didn't seem as useful to users.  We should have plenty of time to revisit it if the fact that it is actually a soft limit is confusing for users.

Reply:
Ok, works for me.